src/component-library/TokenField/TokenField.tsx

```diff
@@ -1,20 +1,29 @@
 import * as React from 'react';
 
-import { TokenFieldLabel, TokenFieldWrapper, TokenFieldInput, TokenFieldUSD } from './TokenField.style';
+import { TokenFieldWrapper, TokenFieldInnerWrapper, TokenFieldSymbol, TokenFieldInput, TokenFieldUSD } from './TokenField.style';
 import { NumberInputProps } from 'component-library/NumberInput';
+import { TokenBalance } from 'component-library/TokenBalance';
 
 interface TokenFieldProps extends NumberInputProps {
-  label: string;
-  approxUSD: string;
+  balance?: string;
+  tokenSymbol: string;
+  valueInUSD: string;
 }
 
 const TokenField = React.forwardRef<HTMLInputElement, TokenFieldProps>(
-  ({ id, label, approxUSD, ...rest }, ref): JSX.Element => {
+  ({ tokenSymbol, valueInUSD, balance, ...rest }, ref): JSX.Element => {
+
     return (
       <TokenFieldWrapper>
-        <TokenFieldInput ref={ref} id={id} {...rest} />
-        <TokenFieldLabel>{label}</TokenFieldLabel>
-        <TokenFieldUSD>{`≈ $ ${approxUSD}`}</TokenFieldUSD>
+        {balance && (
+          // TODO: replace `valueInUSD` value with the balance value in USD according to the underlying token symbol
+          <TokenBalance tokenSymbol={tokenSymbol} value={balance} valueInUSD={`${balance}00`} />
```

Comment:
I think the USD value of balance could be passed into the component as prop the same way the balance is. Or is there a reason for not doing it now and leaving it for later (hence the TODO comment)? :)

Reply:
I thought of passing it as a prop but on a second thought, it could be implemented within the component since the underlying token symbol can be used to obtain the token price. Which one do we prefer? cc @tomjeatt

Reply:
hi – good question. I'd like to calculate the value outside of the UI component (i.e. call `getUsdAmount` in the consuming component) and then pass in the dollar value as either a number or a string. I'm happy with either passing in the `$` symbol as a prop or appending it with a string literal (as you've done here) since it's very specifically a USD amount.

Reply:
As a general convention, we should be doing as little as possible in the UI components in terms of calling utilities etc. In an ideal world (although it may not always be possible) we'd only ever pass strings and numbers into the UI components.